clients/admin-ui/src/features/datastore-connections/add-connection/forms/ConnectorParametersForm.tsx

```diff
@@ -188,32 +193,16 @@ const ConnectorParametersForm: React.FC<ConnectorParametersFormProps> = ({
   );
 
   const getInitialValues = () => {
+    const initialValues = { ...defaultValues };
     if (connection?.key) {
-      defaultValues.name = connection.name;
-      defaultValues.description = connection.description as string;
-      defaultValues.instance_key =
+      initialValues.name = connection.name;
+      initialValues.description = connection.description as string;
+      initialValues.instance_key =
         connection.connection_type === ConnectionType.SAAS
           ? (connection.saas_config?.fides_key as string)
           : connection.key;
-      Object.entries(data.properties).forEach((key) => {
-        // eslint-disable-next-line no-nested-ternary
-        defaultValues[key[0]] = key[1].default
-          ? key[1].default
-          : key[1].type === "integer"
-          ? 0
-          : "";
-      });
-    } else {
-      Object.entries(data.properties).forEach((key) => {
-        // eslint-disable-next-line no-nested-ternary
-        defaultValues[key[0]] = key[1].default
-          ? key[1].default
-          : key[1].type === "integer"
-          ? 0
-          : "";
-      });
     }
-    return defaultValues;
+    return fillInDefaults(initialValues, data);
```

Comment:
basically an effort to remove duplicate code (note the two `Object.entries()...` that are the same above) and also to extract to a function that is tested